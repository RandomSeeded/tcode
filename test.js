import { encode, decode, stream } from './index'

test('encoding an array of null returns expected value', () => {
  expect(stream.streamEncode([null])).toMatchSnapshot();
});

test('decoding an array with type null works', () => {
  expect(stream.streamDecode('~\n', 0, 1)).toMatchSnapshot();
});

test('decoding an array with multiple elements works', () => {
  expect(stream.streamDecode('~\n~\n', 0, 2)).toMatchSnapshot();
});

test('we can both encode and decode', () => {
  const encoded = stream.streamEncode(['abcd', null, 'efgh']);
  expect(encoded).toMatchSnapshot();
  const decoded = stream.streamDecode(encoded, 0, 3);
  expect(decoded).toMatchSnapshot();
});

test('strings without newlines encoded as simple strings', () => {
  expect(encode('foo')).toMatchSnapshot();
});

test('empty string encoded as simple string', () => {
  expect(encode('')).toMatchSnapshot();
});

test('presence of newline switches to bulk string encoding', () => {
  expect(encode('foo\n')).toMatchSnapshot();
});

test('decode null', () => {
  expect(decode('~\n')).toMatchSnapshot();
});

test('decode simple string', () => {
  expect(decode('+foo\n')).toMatchSnapshot();
});

test('decode bulk string', () => {
  expect(decode('$4\nbar\n\n')).toMatchSnapshot();
});

test('decode stream', () => {
  let stream = '+foo\n+bar\n~\n';
  let offset, v0, v1, v2;
  [v0, offset] = decode(stream, offset);
  [v1, offset] = decode(stream, offset);
  [v2, offset] = decode(stream, offset);
  expect([v0, v1, v2]).toMatchSnapshot();
});
