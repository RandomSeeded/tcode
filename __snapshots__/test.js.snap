// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode bulk string 1`] = `
Array [
  "bar",
  7,
]
`;

exports[`decode null 1`] = `
Array [
  null,
  2,
]
`;

exports[`decode simple string 1`] = `
Array [
  "+foo",
  5,
]
`;

exports[`decode stream 1`] = `
Array [
  "+foo",
  "+bar",
  null,
]
`;

exports[`decoding an array with multiple elements works 1`] = `
Array [
  Array [
    null,
    null,
  ],
  4,
]
`;

exports[`decoding an array with type null works 1`] = `
Array [
  Array [
    null,
  ],
  2,
]
`;

exports[`empty string encoded as simple string 1`] = `
"+
"
`;

exports[`encoding an array of null returns expected value 1`] = `
"~
"
`;

exports[`presence of newline switches to bulk string encoding 1`] = `
"$4
foo

"
`;

exports[`strings without newlines encoded as simple strings 1`] = `
"+foo
"
`;

exports[`we can both encode and decode 1`] = `
"+abcd
~
+efgh
"
`;

exports[`we can both encode and decode 2`] = `
Array [
  Array [
    null,
  ],
  14,
]
`;
